cmd/admin/key.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"os/user"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/letsencrypt/boulder/core"
+	"github.com/letsencrypt/boulder/privatekey"
+	sapb "github.com/letsencrypt/boulder/sa/proto"
+)
+
+// subcommandBlockKey encapsulates the "admin block-key" command.
+func (a *admin) subcommandBlockKey(ctx context.Context, args []string) error {
+	subflags := flag.NewFlagSet("block-key", flag.ExitOnError)
+	privKey := subflags.String("private-key", "", "Block issuance for the pubkey corresponding to this private key")
+	comment := subflags.String("comment", "", "Additional context to add to database comment column")
+	_ = subflags.Parse(args)
+
+	if *privKey == "" {
+		return errors.New("the -private-key flag is required")
+	}
+
+	spkiHash, err := a.spkiHashFromPrivateKey(*privKey)
```

Comment:
It seems like it would be easy and possibly useful to allow specifying the SPKI hash directly, in case I have that and not the private-key (and don't have a cert to use with revoke-cert)

Reply:
This is https://github.com/letsencrypt/boulder/issues/7267, not doing it as part of this PR (since I'm not adding any truly new functionality here, just unifying existing functionality).